nessie/src/test/java/org/apache/iceberg/nessie/TestCatalogBranch.java

```diff
@@ -0,0 +1,84 @@
+/*
+ * Licensed to the Apache Software Foundation (ASF) under one
+ * or more contributor license agreements.  See the NOTICE file
+ * distributed with this work for additional information
+ * regarding copyright ownership.  The ASF licenses this file
+ * to you under the Apache License, Version 2.0 (the
+ * "License"); you may not use this file except in compliance
+ * with the License.  You may obtain a copy of the License at
+ *
+ *   http://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing,
+ * software distributed under the License is distributed on an
+ * "AS IS" BASIS, WITHOUT WARRANTIES OR CONDITIONS OF ANY
+ * KIND, either express or implied.  See the License for the
+ * specific language governing permissions and limitations
+ * under the License.
+ */
+
+package org.apache.iceberg.nessie;
+
+import com.dremio.nessie.client.NessieClient;
+import com.dremio.nessie.error.NessieConflictException;
+import com.dremio.nessie.error.NessieNotFoundException;
+import com.dremio.nessie.model.Branch;
+import org.apache.iceberg.Table;
+import org.apache.iceberg.catalog.TableIdentifier;
+import org.apache.iceberg.types.Types;
+import org.junit.Assert;
+import org.junit.Test;
+
+public class TestCatalogBranch extends BaseTestIceberg {
+
+  public TestCatalogBranch() {
+    super("main");
+  }
+
+  @SuppressWarnings("VariableDeclarationUsageDistance")
+  @Test
+  public void testBasicBranch() throws NessieNotFoundException, NessieConflictException {
+    TableIdentifier foobar = TableIdentifier.of("foo", "bar");
+    TableIdentifier foobaz = TableIdentifier.of("foo", "baz");
+    Table bar = createTable(foobar, 1); // table 1
+    createTable(foobaz, 1); // table 2
+    catalog.refresh();
+    createBranch("test", catalog.currentHash());
+
+    hadoopConfig.set(NessieClient.CONF_NESSIE_REF, "test");
+
+    NessieCatalog newCatalog = initCatalog("test");
+    String initialMetadataLocation = metadataLocation(newCatalog, foobar);
+    Assert.assertEquals(initialMetadataLocation, metadataLocation(catalog, foobar));
+    Assert.assertEquals(metadataLocation(newCatalog, foobaz), metadataLocation(catalog, foobaz));
```

Comment:
Context helpers would make this test more readable and would be helpful if it ever fails.

Reply:
ive re-worked this test class. Let me know if thats easier to read

Reply:
Looks a lot better!